Userland/Applications/Calculator/KeypadValue.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "KeypadValue.h"
+#include <AK/Math.h>
+#include <AK/String.h>
+
+KeypadValue::KeypadValue(i64 value, u8 decimal_places)
+    : m_value(value)
+    , m_decimal_places(decimal_places)
+{
+}
+
+KeypadValue::KeypadValue(i64 value)
+    : m_value(value)
+{
+}
+
+KeypadValue KeypadValue::operator+(KeypadValue const& rhs)
+{
+    return operate_on_keypadvalues<KeypadValue>(*this, rhs, [](KeypadValue const&, KeypadValue const& more_decimal_places, i64 less_decimal_places_equalized, i64 more_decimal_places_equalized, bool) -> KeypadValue {
+        return {
+            more_decimal_places_equalized + less_decimal_places_equalized,
+            more_decimal_places.m_decimal_places
+        };
+    });
+}
+
+KeypadValue KeypadValue::operator-(KeypadValue const& rhs)
+{
+    return *this + (-rhs); // NOTE: dirty little shorthand
```

Comment:
The `NOTE` doesn't seem particularly useful, unless you meant it to be treated as a FIXME?